portal/src/graphql/adminapi/UserDetailsConnectedIdentities.tsx

```diff
@@ -230,22 +296,79 @@ const UserDetailsConnectedIdentities: React.FC<UserDetailsConnectedIdentitiesPro
     };
   }, [locale, identities]);
 
+  const onRemoveClicked = useCallback(
+    (identityID: string, identityName: string) => {
+      setConfirmationDialogData({
+        identityID,
+        identityName,
+      });
+    },
+    [setConfirmationDialogData]
+  );
+
+  const onDismissConfirmationDialog = useCallback(() => {
+    setConfirmationDialogData(null);
+  }, []);
+
+  const onConfirmRemoveIdentity = useCallback(() => {
+    if (confirmationDialogData == null) {
+      return;
+    }
+    const { identityID } = confirmationDialogData;
+    deleteIdentity(identityID)
+      .then((success) => {
+        if (success) {
+          refetchUserDetail();
```

Comment:
Instead of refetching manually after a mutation, I prefer using `refetchQueries` of mutation options to automatically refetch queries to decouple dependencies between queries and mutations from presentational concerns.